ldclient_end_to_end_fdv2_test.go

```diff
@@ -0,0 +1,228 @@
+	})
+}
+
+func TestFDV2StreamingSynchronizerUsesCustomTLSConfiguration(t *testing.T) {
+	data := ldservicesv2.NewServerSDKData().Flags(alwaysTrueFlag)
+
+	protocol := ldservicesv2.NewStreamingProtocol().
+		WithIntent(fdv2proto.ServerIntent{Payload: fdv2proto.Payload{
+			ID: "fake-id", Target: 0, Code: "xfer-full", Reason: "payload-missing",
+		}}).
+		WithPutObjects(data.ToPutObjects()).
+		WithTransferred(1)
+
+	streamHandler, streamSender := ldservices.ServerSideStreamingServiceHandler(protocol.Next())
+	protocol.Enqueue(streamSender)
+
+	httphelpers.WithSelfSignedServer(streamHandler, func(server *httptest.Server, certData []byte, certs *x509.CertPool) {
+
+		config := Config{
+			Events:  ldcomponents.NoEvents(),
+			HTTP:    ldcomponents.HTTPConfiguration().CACert(certData),
+			Logging: ldcomponents.Logging().Loggers(sharedtest.NewTestLoggers()),
+			DataSystem: ldcomponents.DataSystem().WithEndpoints(
+				ldcomponents.Endpoints{Streaming: server.URL}).Streaming(),
+		}
+
+		client, err := MakeCustomClient(testSdkKey, config, time.Second*50000)
+		require.NoError(t, err)
+		defer client.Close()
+
+		value, _ := client.BoolVariation(alwaysTrueFlag.Key, testUser, false)
+		assert.True(t, value)
+	})
+}
+
+func TestFDV2StreamingSynchronizerTimesOut(t *testing.T) {
+	data := ldservicesv2.NewServerSDKData().Flags(alwaysTrueFlag)
+
+	protocol := ldservicesv2.NewStreamingProtocol().
+		WithIntent(fdv2proto.ServerIntent{Payload: fdv2proto.Payload{
+			ID: "fake-id", Target: 0, Code: "xfer-full", Reason: "payload-missing",
+		}}).
+		WithPutObjects(data.ToPutObjects()).
+		WithTransferred(1)
+
+	streamHandler, streamSender := ldservices.ServerSideStreamingServiceHandler(protocol.Next())
+	protocol.Enqueue(streamSender)
+
+	slowHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(300 * time.Millisecond)
+		streamHandler.ServeHTTP(w, r)
+	})
+
+	httphelpers.WithServer(slowHandler, func(streamServer *httptest.Server) {
+		logCapture := ldlogtest.NewMockLog()
+
+		config := Config{
+			Events:           ldcomponents.NoEvents(),
+			Logging:          ldcomponents.Logging().Loggers(logCapture.Loggers),
+			ServiceEndpoints: interfaces.ServiceEndpoints{Streaming: streamServer.URL},
+			DataSystem: ldcomponents.DataSystem().WithEndpoints(
+				ldcomponents.Endpoints{Streaming: streamServer.URL}).Streaming(),
+		}
+
+		client, err := MakeCustomClient(testSdkKey, config, time.Millisecond*100)
+		require.Error(t, err)
+		require.NotNil(t, client)
+		defer client.Close()
+
+		assert.Equal(t, "timeout encountered waiting for LaunchDarkly client initialization", err.Error())
+
+		value, _ := client.BoolVariation(alwaysTrueFlag.Key, testUser, false)
+		assert.False(t, value)
+
+		assert.Equal(t, []string{"Timeout encountered waiting for LaunchDarkly client initialization"}, logCapture.GetOutput(ldlog.Warn))
+		assert.Len(t, logCapture.GetOutput(ldlog.Error), 0)
+	})
+}
```

Comment:
TODO: we should have a test that sets up a polling handler and a streaming handler, and verifies that the `DataSystem().Default()` config uses two-phase init.